src-cpp/rdkafkacpp.h

```diff
@@ -2931,6 +2931,31 @@ class RD_EXPORT KafkaConsumer : public virtual Handle {
    */
   virtual Error *incremental_unassign(
       const std::vector<TopicPartition *> &partitions) = 0;
+
+  /**
+   * @brief Close and shut down the consumer.
+   *
+   * Same behaviour as RdKafka::KafkaConsumer::close() but with the following
+   * changes:
+   *
+   * Rebalance events/callbacks (etc) will be forwarded to the
+   * application-provided \p queue. The application must poll this queue (in
+   * another thread) until this close function returns or
+   * RdKafka::KafkaConsumer::closed() returns true.
+   *
+   * @remark Depending on consumer group join state there may or may not be
+   *         rebalance events emitted on \p rkqu.
+   *
+   * @returns an error object if the consumer close failed, else NULL.
+   *
+   * @sa RdKafka::KafkaConsumer::closed()
+   */
+  virtual Error *close(Queue *queue) = 0;
+
+
+  /** @returns true if the consumer has been closed with
+   *           RdKafka::KafkaConsumer::close(), else false. */
+  virtual bool closed() = 0;
```

Comment:
i was thinking it would be more natural to have another event type for this (is this a queue based API or not!?), but this works, and in practice is probably more ergonomic i think.